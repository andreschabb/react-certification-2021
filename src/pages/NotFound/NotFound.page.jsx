import React from 'react';
import { Link } from 'react-router-dom';

import { Container, HomeLink, Title } from './NotFound.styles';

const NotFoundPage = () => {
  return (
    <Container>
      <Title>Page Not Found</Title>
      <img data-testid="gif" src="404.gif" alt="Page Not Found" />
      <HomeLink>
        <Link data-testid="go-home" to="/" className="home-link">
          Go back home
        </Link>
      </HomeLink>
    </Container>
  );
};

export default NotFoundPage;
