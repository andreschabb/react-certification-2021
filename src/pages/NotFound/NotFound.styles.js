import styled from 'styled-components';

export const Container = styled.section`
  text-align: center;
  height: 100%;
  width: 40%;
  margin-bottom: 5%;
`;

export const Title = styled.h1`
  font-size: 4rem;
  letter-spacing: -6px;
  font-family: 'fantasy';
  font-weight: normal;

  @media screen and (max-width: 650px) {
    font-size: 3rem;
    letter-spacing: -4px;
  }

  @media screen and (max-width: 425px) {
    font-size: 2rem;
    letter-spacing: -2px;
  }
`;

export const HomeLink = styled.div`
  a.home-link {
    font-size: 1rem;
    text-decoration: none;
    display: block;
    text-align: center;
    padding: 2rem;
    color: black;

    &:before {
      content: '‚Üê';
      padding-right: 0.2rem;
      display: inline-block;
    }

    &:hover {
      color: red;
    }
  }
`;
