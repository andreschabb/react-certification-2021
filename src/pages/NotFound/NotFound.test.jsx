import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

import NotFound from './NotFound.page';

describe('NotFound', () => {
  it('should renders', () => {
    render(
      <MemoryRouter initialEntries={['/notfound']}>
        <NotFound />
      </MemoryRouter>
    );

    expect(screen.getByText('Page Not Found')).toBeTruthy();
    expect(screen.getByTestId('go-home').getAttribute('href')).toBe('/');
    expect(screen.getByTestId('gif').getAttribute('alt')).toBe('Page Not Found');
  });
});
