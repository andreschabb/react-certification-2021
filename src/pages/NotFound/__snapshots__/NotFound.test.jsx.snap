// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should renders 1`] = `
<div>
  <section
    class="sc-bdfBwQ dDpctk"
  >
    <h1
      class="sc-gsTCUz ifuuBC"
    >
      Page Not Found
    </h1>
    <img
      alt="Page Not Found"
      data-testid="gif"
      src="404.gif"
    />
    <div
      class="sc-dlfnbm cWLLzh"
    >
      <a
        class="home-link"
        data-testid="go-home"
        href="/"
      >
        Go back home
      </a>
    </div>
  </section>
</div>
`;
